src/prefect/cli/deployment.py

```diff
@@ -873,6 +873,16 @@ async def run(
                 f"deployment: {listrepr(unknown_keys, sep=', ')}"
                 f"\n{available_parameters}"
             )
+        parameters = {**(deployment.parameters or {}), **(parameters or {})}
+        if flow_run_name:
+            try:
+                flow_run_name = flow_run_name.format(**parameters)
```

Comment:
I don't think we want to overwrite the value of `parameters` here since that could have the side effect of overriding the default parameters set on the deployment. 
```suggestion
        templating_parameters = {**(deployment.parameters or {}), **(parameters or {})}
        if flow_run_name:
            try:
                flow_run_name = flow_run_name.format(**templating_parameters)
```